### PR TITLE
Add models api

### PR DESCRIPTION
**Pull request by LielinJiang**

### CODE REVIEW

File: image_classification/imagenet_dataset.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    def process(sample):
+        img, label = sample
+        for fn in functions:
+            img = fn(img)
+        return img, label
+
+    return process
+
+
+def image_folder(path):
```
Comment (qingqing01) [inline image_classification/imagenet_dataset.py]: image_folder, need to be a common interface

File: models/resnet.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+        x = fluid.layers.elementwise_add(x=short, y=conv2)
+
+        layer_helper = LayerHelper(self.full_name(), act='relu')
+        return layer_helper.append_activation(x)
+        # return fluid.layers.relu(x)
+
+
+class ResNet(Model):
+    def __init__(self, Block, depth=50, num_classes=1000):
```
Comment (qingqing01) [inline models/resnet.py]: 1. Need to consider whether to fix parameter name, but allow to add a prefix
2. Need to consider how to get endpoint at each stage

Comment (qingqing01) [inline models/resnet.py]: 1. Need to consider whether to fix parameter name, but allow to add a prefix
2. Need to consider how to get endpoint at each stage

File: image_classification/imagenet_dataset.py (hunk 1 above)
Comment (LielinJiang) [reply image_classification/imagenet_dataset.py]: done, thanks.

File: models/resnet.py (hunk 2 above)
Comment (LielinJiang) [reply models/resnet.py]: this is an example how to add a prefix and how to get endpoint.
image

image

File: models/resnet.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                initializer=fluid.initializer.Uniform(-stdv, stdv)))
+
+    def forward(self, inputs):
+        x = self.conv(inputs)
+        x = self.pool(x)
+        for layer in self.layers:
+            x = layer(x)
+        x = self.global_pool(x)
+        x = fluid.layers.reshape(x, shape=[-1, self.fc_input_dim])
+        x = self.fc(x)
```
Comment (qingqing01) [inline models/resnet.py:161]: fc is not needed in some model.

File: models/resnet.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+def resnet50(pretrained=False):
+    return _resnet('resnet50', BottleneckBlock, 50, pretrained)
+
+
+def resnet101(pretrained=False):
+    return _resnet('resnet101', BottleneckBlock, 101, pretrained)
+
+
+def resnet152(pretrained=False):
+    return _resnet('resnet152', BottleneckBlock, 152, pretrained)
```
Comment (qingqing01) [inline models/resnet.py:187]: Also needs resnet18, 34

Comment (LielinJiang) [reply models/resnet.py:187]: ok, resnet18 and resnet34 will be added in next pr

**Pull request merged**